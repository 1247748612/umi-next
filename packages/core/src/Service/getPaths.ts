import { join } from 'path';
import { existsSync, statSync } from 'fs';

function isDirectoryAndExist(path: string) {
  return existsSync(path) && statSync(path).isDirectory();
}

export default function({
  cwd,
  config,
  env,
}: {
  cwd: string;
  config: any;
  env: string | undefined;
}) {
  let absSrcPath = cwd;
  if (isDirectoryAndExist(join(cwd, 'src'))) {
    absSrcPath = join(cwd, 'src');
  }
  const absPagesPath = config.singular
    ? join(absSrcPath, 'page')
    : join(absSrcPath, 'pages');

  return {
    cwd,
    absNodeModulesPath: join(cwd, 'node_modules'),
    absOutputPath: join(cwd, config.outputPath || './dist'),
    absSrcPath,
    absPagesPath,
    absTmpPath: join(
      absSrcPath,
      ['.umi', env !== 'development' && env].filter(Boolean).join('-'),
    ),
  };
}
