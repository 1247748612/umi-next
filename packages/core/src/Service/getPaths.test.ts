import { join } from 'path';
import getPaths from './getPaths';

const fixtures = join(__dirname, 'fixtures');

function stripCwd(obj: object, cwd: string) {
  return Object.keys(obj).reduce((memo, key) => {
    memo[key] = obj[key].replace(cwd, '.');
    return memo;
  }, {});
}

test('empty', () => {
  const cwd = join(fixtures, 'getPaths-empty');
  expect(
    stripCwd(
      getPaths({
        cwd,
        config: {},
        env: 'development',
      }),
      cwd,
    ),
  ).toMatchSnapshot();
});

test('empty production', () => {
  const cwd = join(fixtures, 'getPaths-empty');
  expect(
    stripCwd(
      getPaths({
        cwd,
        config: {},
        env: 'production',
      }),
      cwd,
    ),
  ).toMatchSnapshot();
});

test('empty config singular', () => {
  const cwd = join(fixtures, 'getPaths-empty');
  expect(
    stripCwd(
      getPaths({
        cwd,
        config: {
          singular: true,
        },
        env: 'development',
      }),
      cwd,
    ),
  ).toMatchSnapshot();
});

test('empty config outputPath', () => {
  const cwd = join(fixtures, 'getPaths-empty');
  expect(
    stripCwd(
      getPaths({
        cwd,
        config: {
          outputPath: './www',
        },
        env: 'development',
      }),
      cwd,
    ),
  ).toMatchSnapshot();
});

test('src', () => {
  const cwd = join(fixtures, 'getPaths-with-src');
  expect(
    stripCwd(
      getPaths({
        cwd,
        config: {},
        env: 'development',
      }),
      cwd,
    ),
  ).toMatchSnapshot();
});

test('src config singular', () => {
  const cwd = join(fixtures, 'getPaths-with-src');
  expect(
    stripCwd(
      getPaths({
        cwd,
        config: {
          singular: true,
        },
        env: 'development',
      }),
      cwd,
    ),
  ).toMatchSnapshot();
});
