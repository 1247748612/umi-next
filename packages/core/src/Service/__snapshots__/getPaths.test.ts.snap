// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty 1`] = `
Object {
  "absNodeModulesPath": "./node_modules",
  "absOutputPath": "./dist",
  "absPagesPath": "./pages",
  "absSrcPath": ".",
  "absTmpPath": "./.umi",
  "cwd": ".",
}
`;

exports[`empty config outputPath 1`] = `
Object {
  "absNodeModulesPath": "./node_modules",
  "absOutputPath": "./www",
  "absPagesPath": "./pages",
  "absSrcPath": ".",
  "absTmpPath": "./.umi",
  "cwd": ".",
}
`;

exports[`empty config singular 1`] = `
Object {
  "absNodeModulesPath": "./node_modules",
  "absOutputPath": "./dist",
  "absPagesPath": "./page",
  "absSrcPath": ".",
  "absTmpPath": "./.umi",
  "cwd": ".",
}
`;

exports[`empty production 1`] = `
Object {
  "absNodeModulesPath": "./node_modules",
  "absOutputPath": "./dist",
  "absPagesPath": "./pages",
  "absSrcPath": ".",
  "absTmpPath": "./.umi-production",
  "cwd": ".",
}
`;

exports[`src 1`] = `
Object {
  "absNodeModulesPath": "./node_modules",
  "absOutputPath": "./dist",
  "absPagesPath": "./src/pages",
  "absSrcPath": "./src",
  "absTmpPath": "./src/.umi",
  "cwd": ".",
}
`;

exports[`src config singular 1`] = `
Object {
  "absNodeModulesPath": "./node_modules",
  "absOutputPath": "./dist",
  "absPagesPath": "./src/page",
  "absSrcPath": "./src",
  "absTmpPath": "./src/.umi",
  "cwd": ".",
}
`;
